Jenkinsfile

```diff
@@ -13,13 +13,31 @@ pipeline{
 			steps {
 				sh "sudo DEBIAN_FRONTEND=noninteractive apt-get install -yq build-essential mtools libssl-dev pkg-config"
 				sh "sudo apt-get install -yq flex bison libelf-dev qemu-utils qemu-system libglib2.0-dev libpixman-1-dev libseccomp-dev socat"
+				sh "sudo snap install docker"
 			}
 		}
 		stage ('Install Rust') {
 			steps {
 				sh "nohup curl https://sh.rustup.rs -sSf | sh -s -- -y"
 			}
 		}
+		stage ('Install Cargo utilities') {
```

Comment:
Just move these steps into `scripts/run_cargo_tests` after sourcing the cargo env to resolve the build error.